packages/console/src/pages/EnterpriseSso/Guide/XmlFileUploader/index.tsx

```diff
@@ -0,0 +1,73 @@
+import classNames from 'classnames';
+import { useRef, type ChangeEvent } from 'react';
+
+import Delete from '@/assets/icons/delete.svg';
+import FileIcon from '@/assets/icons/file-icon.svg';
+import UploaderIcon from '@/assets/icons/upload.svg';
+import Button from '@/ds-components/Button';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  onChange: (file?: File) => void;
+  value?: File;
+};
+
+function XmlFileUploader({ onChange, value }: Props) {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      onChange(file);
+    }
+  };
+
+  const handleRemove = () => {
+    onChange(undefined);
+  };
+
+  return (
+    <div>
+      {value ? (
+        <div className={styles.preview}>
+          <FileIcon />
+          <div className={styles.fileInfo}>
+            <span className={styles.fileName}>{value.name}</span>
+            <span className={styles.fileSize}>{`${(value.size / 1024).toFixed(2)} KB`}</span>
+          </div>
+          <button
+            type="button"
+            className={classNames(styles.delete)}
+            onClick={() => {
+              handleRemove();
+            }}
+          >
+            <Delete className={styles.icon} />
+          </button>
+        </div>
+      ) : (
+        <div>
+          <Button
+            icon={<UploaderIcon />}
+            title="enterprise_sso_details.upload_idp_metadata_button_text"
+            size="large"
+            onClick={() => {
+              inputRef.current?.click();
+            }}
+          />
+          <input
+            ref={inputRef}
+            value={value}
+            type="file"
+            accept=".xml"
+            className={styles.fileInput}
+            onChange={handleFileChange}
+          />
+        </div>
+      )}
+    </div>
```

Comment:
Do we have a max file size limit for the xml files?

Reply:
Good call, can add a file size limit as 500KB for now. Normal metadata XML file should not exceed this limit.